File: gfg-weekly-coding-contest-110/Remove the balls.py
from typing import List

class Solution:
    # Version 1: Stack
    # TC: O(n), SC: O(n)
    def finLength(self, N : int, color : List[int], radius : List[int]) -> int:
        # code here
        stack = []
        for i in range(N):
            if stack and color[stack[-1]] == color[i] and radius[stack[-1]] == radius[i]:
                stack.pop()
            else:
                stack.append(i)
        return len(stack)

File: gfg-weekly-coding-contest-110/Remove the balls.java


class Solution {
    // Version 1: Stack
    // TC: O(n), SC: O(n)
    public static int finLength(int N, int[] color, int[] radius) {
        // code here
        Stack<Integer> st = new Stack<Integer>();
        for (int i = 0; i < N; i++) {
            if (!st.empty() && color[st.peek()] == color[i] && radius[st.peek()] == radius[i]) {
                st.pop();
            } else {
                st.push(i);
            }
        }
        return st.size();
    }
}
        